python/GafferDispatchUI/LocalJobs.py

```diff
@@ -0,0 +1,365 @@
+
+		result = Gaffer.Path.property( self, name )
+		if result is not None :
+			return result
+
+		if self.__job is None :
+			return None
+
+		if name == "localDispatcher:status" :
+			return self.__job.status().name
+		elif name == "localDispatcher:id" :
+			return self.__job.id()
+		elif name == "localDispatcher:jobName" :
+			return self.__job.name()
+		elif name == "localDispatcher:directory" :
+			return self.__job.directory()
+		elif name == "localDispatcher:startTime" :
+			return self.__job.startTime()
+		elif name == "localDispatcher:runningTime" :
+			return self.__job.runningTime().total_seconds()
+		elif name == "localDispatcher:cpuUsage" :
+			return self.__job.cpuUsage()
+		elif name == "localDispatcher:memoryUsage" :
+			return self.__job.memoryUsage()
+
+		return None
+
+	def job( self ) :
+
+		return self.__job
+
+	def jobPool( self ) :
+
+		return self.__jobPool
+
+	def isLeaf( self, canceller = None ) :
+
+		return len( self )
+
+	def _children( self, canceller ) :
+
+		if self.isLeaf() :
+			return []
+
+		result = []
+		for index, job in enumerate( self.__jobPool.jobs() ) :
+			result.append(
+				_LocalJobsPath(
+					jobPool = self.__jobPool,
+					job = job,
+					path = [ str( index ) ],
+				)
+			)
+
+		return result
+
+class _StatusColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		status = path.property( "localDispatcher:status", canceller )
+		return GafferUI.PathColumn.CellData(
+			icon = f"localDispatcherStatus{status}.png",
+			toolTip = status,
+			sortValue = int( GafferDispatch.LocalDispatcher.Job.Status[status] )
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "Status" )
+
+class _RunningTimeColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		seconds = int( path.property( "localDispatcher:runningTime", canceller ) )
+
+		return GafferUI.PathColumn.CellData(
+			value = "{:02}:{:02}:{:02}".format(
+				seconds // 3600,
+				(seconds % 3600) // 60,
+				seconds % 60
+			),
+			sortValue = seconds
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "Running Time" )
+
+class _CPUUsageColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		cpu = path.property( "localDispatcher:cpuUsage", canceller )
+
+		return GafferUI.PathColumn.CellData(
+			value = f"{cpu:.2f}" if cpu is not None else "---",
+			sortValue = cpu if cpu is not None else 0.0,
+			toolTip = "CPU usage for current batch"
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "CPU" )
+
+class _MemoryUsageColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		memory = path.property( "localDispatcher:memoryUsage", canceller )
+
+		return GafferUI.PathColumn.CellData(
+			value = "{:.2f}GB".format( memory / (1024 ** 3) ) if memory is not None else "---",
+			sortValue = memory if memory is not None else 0,
```

Comment:
I've seen this lead to `OverflowError: bad numeric conversion: positive overflow` on batches with memory usage above 2 gigabytes, maybe we should sort by the remapped to gigabytes value instead?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5557/commits/97b829a7ef9fff7ae75181ccc0992d10c97cc4ad by returning UInt64Data explicitly rather than relying on the implicit conversion to IntData that was overflowing.